scheduler/src/cook/mesos/api.clj

```diff
@@ -200,11 +203,31 @@
   [s]
   (re-matches #"[\.a-zA-Z0-9_-]{1,128}" s))
 
+(defn non-empty-max-128-characters
+  [s]
+  (< 0 (.length s) 129))
+
+(defn date?
+  [s]
+  (re-matches #"[0-9]{8}" s))
+
 (def Application
   "Schema for the application a job corresponds to"
   {:name (s/constrained s/Str non-empty-max-128-characters-and-alphanum?)
    :version (s/constrained s/Str non-empty-max-128-characters-and-alphanum?)})
 
+(def Dataset
```

Comment:
Maybe elaborate on the forest level view of this:

 " For locality purposes, we measure locality in terms of  'datasets which can optionally be broken up into a set of associated partitions', a partition, if it exists is a date range. A dataset name is a dictionary with key and value strings of at most 128 characters"

This name seems a bit confusing? Is a 'dataset', the name of the entire locality-key (the datasetname + partitions) or just the dataset name? Should we rename this Datasetname? Or rename :datasets (on line 302, below) to something like 'localitykey'?

Reply:
Did this get resolved?

Reply:
@scrosby

Reply:
I'm fine if you want to keep the name, but please put in explanatory text. E.g, something like the text block draft I wrote above, with whatever corrections are needed to make it correct.

Reply:
Done